### PR TITLE
View user profile

### LINKED ISSUE #37: View User Profile

### PR DESCRIPTION
**Pull request by carriezhengjr**
Fixes #37 

Currently, Languages shows "None" because languages are not specified when user creates the account.
Guess we can add more stuff like analytics later below, so the page won't look so empty.

Light mode:
Screenshot 2023-09-20 at 22 15 48

Dark mode:
Screenshot 2023-09-20 at 22 15 56

### CODE REVIEW

**Review by sltsheryl — COMMENTED** [review]
currently, the existing profile works but i'm thinking instead of putting the avatar on the navigation bar, we can do like the structure below?

i think we can have multiple panels too so following leetcode, we can have 
1. the profile panel on the left, and 
2. to the right top, number of solves (i think for now can do up the frontend just add in placeholder numbers don't really need to worry abt backend yet), and 
3. the right bottom  github-like analytics (that's part of our nice to have so can just leave it blank first unless you want to do it?)

Screenshot 2023-09-27 at 4 44 04 PM

File: frontend/src/pages/users/ViewProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        setUserData(response);
+      } catch (error) {
+        console.error('Error fetching user profile:', error);
+      }
+    };
+
+    fetchUserProfile().catch((error) => {
+      console.error('Error fetching user profile:', error);
+    });
+  }, []);
```
Comment (seelengxd) [inline frontend/src/pages/users/ViewProfile.tsx]: nit: now that i relook at this - this is possibly unnecessary , you can get the user via

```js
const user = useAppSelector(selectUser)
```

**Comment by seelengxd** [comment]
after adding the structure from the comment above, can you also place the deregister button #58 here as well? (maybe just below the edit button) thanks!

**Comment by carriezhengjr** [comment]
> after adding the structure from the comment above, can you also place the deregister button #58 here as well? (maybe just below the edit button) thanks!

I've added here #67 because I think this PR might be outdated.

Should we get the current one as MVP first? Then I can add those structures after midterms hmmm 🤔...

**Pull request closed without merging**